crates/trident/src/engine/clean_install.rs

```diff
@@ -257,6 +257,10 @@ fn stage_clean_install(
         return Err(original_error).message("Failed to execute in chroot");
     }
 
+    // Update Host Configuration with the paths of extension images.
+    let updated_hc =
+        engine::get_extensions_subsystem(subsystems)?.update_host_configuration(&ctx)?;
```

Comment:
is there a reason to limit this to extensions?  maybe call all subsystem's update_host_configuration (and currently all but extensions is no-op)